src/test/kotlin/com/terraformation/backend/accelerator/DeliverableServiceTest.kt

```diff
@@ -99,18 +99,15 @@ class DeliverableServiceTest : DatabaseTest(), RunsAsUser {
             clock.instant,
             user.userId,
             clock.instant,
-        ),
-        submissionsDao.fetchOneById(id))
+        ))
   }
 
   @Test
   fun `updates existing submission to Complete status`() {
     insertSubmission(deliverableId = deliverable1, submissionStatus = SubmissionStatus.NotNeeded)
-    val existing = submissionsDao.fetchOneById(inserted.submissionId)!!
+    val existing = dslContext.fetchOne(SUBMISSIONS, SUBMISSIONS.ID.eq(inserted.submissionId))!!
     service.setDeliverableCompletion(deliverable1, inserted.projectId, true)
-    assertEquals(
-        existing.copy(submissionStatusId = SubmissionStatus.Completed),
-        submissionsDao.fetchOneById(inserted.submissionId))
+    assertTableEquals(existing.apply { submissionStatusId = SubmissionStatus.Completed })
```

Comment:
Not totally happy with this pattern; it feels like there may be more room for improvement here.

Reply:
Improved this a little bit by using the jOOQ `fetchSingle` method. Still don't love the `apply {}` part of this since it mutates the existing record but I don't see a better alternative.